### DISCUSSION TITLE
I want source to learn React js ?

### DISCUSSION THREAD

**Discussion by musaabdullah** [community/community#75943] [Programming Help]
I need source to learn react js.

**Comment by ghost** [comment]
Hello @musaabdullah 

There are tons of recourses to learn React. You can buy some courses on [Udemy](https://www.udemy.com/?utm_source=adwords-brand&utm_medium=udemyads&utm_campaign=Brand-Udemy_la.EN_cc.ROW&utm_term=_._ag_80315195513_._ad_535757779892_._de_c_._dm__._pl__._ti_kwd-296956216253_._li_1007469_._pd__._&utm_term=_._pd__._kw_udemy_._&matchtype=b&gad_source=1&gclid=Cj0KCQiA3uGqBhDdARIsAFeJ5r3cM_cOHSZpnI-wjnGcqjBl7Sr0Ys-7lJUAHbidMcbXr2d5aJTnIx4aAoqREALw_wcB) (there are good structured react courses that covers most important parts of React).

**However** if you are lloking for free courses then YouTube is one of the best platform for that. I can't suggest any tutorial specifically since there tons of channels for that but I can list few of them:

- [Net Ninja - Full React Tutorial](https://www.youtube.com/watch?v=j942wKiXFu8&list=PL4cUxeGkcC9gZD-Tvwfod2gaISzfRiP9d)
- [CodeVolution - ReactJS Tutorial ](https://www.youtube.com/watch?v=QFaFIcGhPoM&list=PLC3y8-rFHvwgg3vaYJgHGnModB54rxOk3) (It's little bit old but still good to understand basic principles of react)
- [FreeCodeCamp - React Course](https://www.youtube.com/watch?v=bMknfKXIFA8&t=130s) (one long video :smile: )

**Also** you should definitely check [Web Dev Simplify](https://www.youtube.com/@WebDevSimplified) and [Jack Herrington](https://www.youtube.com/@jherr) on youtube. Those are one of the best chanels to learn React (mostlly they have small videos about basics of react, state managments, hooks and so on. so defenitelly worth to check)

**For Practice** you can check [3WSchools](https://www.w3schools.com/react/default.asp) to practice and learn at the same time. 

if you prefer reading over watching videos you can use 3WSchools link included above and also you can check [React Documentation](https://legacy.reactjs.org/docs/getting-started.html). its really helpfull to read documentation as well.

Hope this helps

**Comment by fatmadurak** [comment]


**Reply by queenofcorgis** [reply]
@fatmadurak,

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by Aghaei-Dev** [comment]
two instructor in Udemy 
john smilga
maximilian Schwarzmuller
are great 
i learned from these two person

**Comment by Olenka-Hryk** [comment]
Hi! 
You can find a good course on React at this site: https://egghead.io/courses/the-beginner-s-guide-to-react
Or here: https://react.dev/learn

**Comment by ghost** [comment]
Video Sources for learning React:
1. [Free Code Camp's Project based tutorial](https://youtu.be/bMknfKXIFA8?si=JWouvFb_dRfj3SYj)
2. [Playlist for React](https://youtube.com/playlist?list=PLC3y8-rFHvwgg3vaYJgHGnModB54rxOk3&si=orPnp4-slcnLVZhf)
3. [Project Based Learning](https://youtu.be/F627pKNUCVQ?si=2y_5Bg7qMlUw_MWO)

Web sources for learning React:
1. [W3Schools](https://www.w3schools.com/REACT/react_intro.asp)
2. [Geeks for Geeks](https://www.geeksforgeeks.org/reactjs-tutorials/)

UDEMY courses on React is also good. But these are free sources to learn React. But make sure you also create projects with the knowledge you learnt. 

All the best for web development journey.

Thank you

**Comment by twolama** [comment]
you can get the resources at https://spring.io/guides/tutorials/react-and-spring-data-rest/  then keep trying.

**Comment by sahilatahar** [comment]
I will prefer you to follow official documentation [react.dev](https://react.dev/). If you are facing problem and want explained video then I will recommend you `web dev simplified` and `Dave Grey` react playlist on YouTube and also for for react topic explained blog search on Google `<Topic Name> medium`. We will find docs. For error solution just copy paste error message on Google and 90% of the time you will find solution.

**Comment by gitfullstacker** [comment]
You can refer my repository that I made:
https://github.com/chenxidev1129/Movie-Website

**Comment by dzenebieri** [comment]
https://www.facebook.com/groups/reactge
